ocp_resources/cluster_role.py

```diff
@@ -30,39 +33,14 @@ def __init__(
             delete_timeout=delete_timeout,
             **kwargs,
         )
-        self.api_groups = api_groups
-        self.permissions_to_resources = permissions_to_resources
-        self.verbs = verbs
+        if not rules and not yaml_file:
+            raise ValueError("must send rules or yaml_file")
+        self.rules = rules
         self.desired_state = {"rules": []}
 
     def to_dict(self):
         if not self.res:
             super().to_dict()
 
-        if not self.yaml_file and self.permissions_to_resources:
-            self.add_rule(
-                api_groups=self.api_groups,
-                permissions_to_resources=self.permissions_to_resources,
-                verbs=self.verbs,
-            )
-
-    def add_rule(
-        self,
-        verbs,
-        api_groups=None,
-        permissions_to_resources=None,
-    ):
-        if not self.res:
-            super().to_dict()
-
-        rule = {"verbs": verbs}
-        if api_groups:
-            rule["apiGroups"] = api_groups
-        if permissions_to_resources:
-            rule["resources"] = permissions_to_resources
-        if rule:
-            self._set_rule(rule=rule)
-
-    def _set_rule(self, rule):
-        self.desired_state["rules"].append(rule)
-        self.res["rules"] = self.desired_state["rules"]
+        if not self.yaml_file and self.rules:
```

Comment:
no need to check `self.rules` here, you will have it if you got here .

Reply:
Ack.